mlkem/asm/aarch64/optimize.sh

```diff
@@ -6,20 +6,92 @@ set -e
 TARGET_NAME="Cortex-A55"
 TARGET=Arm_Cortex_A55
 
-echo "* Base multiplication, ${TARGET_NAME}"
+echo "* polyvec_basemul_acc_montgomery_cached, K=2, ${TARGET_NAME}"
+
+cp polyvec_clean.S polyvec_opt.S
+
+slothy-cli Arm_AArch64 $TARGET \
+  polyvec_opt.S -o polyvec_opt.S \
+  -r polyvec_basemul_acc_montgomery_cached_asm_k2_clean,polyvec_basemul_acc_montgomery_cached_asm_k2_opt \
+  -l k2_loop_start \
+  -c sw_pipelining.enabled=true \
+  -c inputs_are_outputs \
+  -c reserved_regs="[x18, sp]" \
+  -c sw_pipelining.minimize_overlapping=False \
+  -c sw_pipelining.allow_post \
+  -c variable_size \
+  -c constraints.stalls_first_attempt=64
+
+echo "* polyvec_basemul_acc_montgomery_cached, K=3, ${TARGET_NAME}"
+
+slothy-cli Arm_AArch64 $TARGET \
+  polyvec_opt.S -o polyvec_opt.S \
+  -r polyvec_basemul_acc_montgomery_cached_asm_k3_clean,polyvec_basemul_acc_montgomery_cached_asm_k3_opt \
+  -l k3_loop_start \
+  -c sw_pipelining.enabled=true \
+  -c inputs_are_outputs \
+  -c reserved_regs="[x18, sp]" \
+  -c sw_pipelining.minimize_overlapping=False \
+  -c sw_pipelining.allow_post \
+  -c variable_size \
+  -c constraints.stalls_first_attempt=64
+
+echo "* polyvec_basemul_acc_montgomery_cached, K=4, ${TARGET_NAME}"
+
+slothy-cli Arm_AArch64 $TARGET \
+  polyvec_opt.S -o polyvec_opt.S \
+  -r polyvec_basemul_acc_montgomery_cached_asm_k4_clean,polyvec_basemul_acc_montgomery_cached_asm_k4_opt \
+  -l k4_loop_start \
+  -c sw_pipelining.enabled=true \
+  -c inputs_are_outputs \
+  -c reserved_regs="[x18, sp]" \
+  -c sw_pipelining.minimize_overlapping=False \
+  -c variable_size \
+  -c sw_pipelining.allow_post \
+  -c constraints.stalls_first_attempt=64
+
+cp poly_clean.S poly_opt.S
+
+echo "* poly_reduce, ${TARGET_NAME}"
 
 slothy-cli Arm_AArch64 $TARGET \
-  poly_clean.S -o poly_opt.S \
+  poly_opt.S -o poly_opt.S \
   -r poly_reduce_asm_clean,poly_reduce_asm_opt \
   -l loop_start \
   -c sw_pipelining.enabled=true \
   -c inputs_are_outputs \
-  -c reserved_regs="[x0--30,v10--v31,sp]" \
+  -c reserved_regs="[x0--30,v8--v15,sp]" \
```

Comment:
Why is this `x0--30`, but `v8-v15`? Looks inconsitent. Is this a SLOTHY bug?

Reply:
I've made it more consistent, but the intention is to forbid the use of callee-saved registers in clean ASM which does not stash them. And the decision of when to stash in clean ASM is based on whether it seems likely that optimized code would need to use them.